Content.Server/Magic/Components/SpellbookComponent.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Threading;
+using Content.Shared.Actions.ActionTypes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype.Dictionary;
+
+namespace Content.Server.Magic;
+
+[RegisterComponent]
+public sealed class SpellbookComponent : Component
+{
+    /// <summary>
+    /// List of spells that this book has.
+    /// </summary>
+    [ViewVariables]
+    public readonly List<ActionType> Spells = new();
+
+    [DataField("worldSpells", customTypeSerializer: typeof(PrototypeIdDictionarySerializer<int, WorldTargetActionPrototype>))]
+    public readonly Dictionary<string, int> WorldSpells = new();
+
+    [DataField("entitySpells", customTypeSerializer: typeof(PrototypeIdDictionarySerializer<int, EntityTargetActionPrototype>))]
+    public readonly Dictionary<string, int> EntitySpells = new();
+
+    [DataField("instantSpells", customTypeSerializer: typeof(PrototypeIdDictionarySerializer<int, InstantActionPrototype>))]
+    public readonly Dictionary<string, int> InstantSpells = new();
+
```

Comment:
Should make a comment that this is only used for initialisation because I went to magicsystem to find out what it was doing.